fling/component/client/cross_domain_client.py

```diff
@@ -0,0 +1,235 @@
+import copy
+import random
+
+import torch
+from torch.utils.data import DataLoader, Dataset
+import torch.nn as nn
+
+from fling.utils import get_optimizer, VariableMonitor, get_weights
+from fling.utils.registry_utils import CLIENT_REGISTRY
+from .client_template import ClientTemplate
+
+
+@CLIENT_REGISTRY.register('cross_domain_client')
+class CrossDomainClient(ClientTemplate):
```

Comment:
Maybe this class can inherit BaseClient. I think there is no difference in function train, test, finetune, preprocess_data. And the __init__ function can be largely simplified by calling super.__init__()